delira/utils/config.py

```diff
@@ -1,13 +1,509 @@
+
+        Warnings
+        --------
+        If the short form of the nested settings is used integers and str
+        can not be differentiated. Therefore, integers should be avoided.
+
+        Examples
+        --------
+        Create simple configuration with nested keys
+        >>> from delira.utils import Config
+        >>> cf = Config()
+        >>> # automatically generates new nested dictionaries
+        >>> cf['first_level.second_level.third_level'] = 1
+        >>> # form access
+        >>> print(cf['first_level.second_level.third_level'])
+        >>> # traditional access
+        >>> print(cf['first_level']['second_level']['third_level'])
+        >>> # entries can also be accessed with dot operator
+        >>> print(cf.first_level.second_level.thirs_level)
+        """
+
+        super().__init__()
+        self.__dict__ = self
+        if dict_like is not None:
+            self.update(dict_like)
+        self.update(kwargs)
+
+    @non_string_warning
+    def __setattr__(self, key, value):
+        """
+        Set attribute in config
+
+        Parameters
+        ----------
+        key : str
+            attribute name
+        value : any
+            attribute value
+
+        """
+        super().__setattr__(key, self._to_config(value))
+
+    @non_string_warning
+    def __setitem__(self, key, value):
+        """
+        Set items inside dict. Supports setting of nested entries by
+        seperating the individual keys with a '.'.
+
+        Parameters
+        ----------
+        key : str
+            key for new value
+        value : any
+            new value
+        """
+        if not isinstance(key, str) or '.' not in key:
+            super().__setitem__(key, value)
+        else:
+            current_level = self
+            keys = key.split(".")
+            final_key = keys.pop(-1)
+            final_dict = self._traverse_keys(keys, create=True)
+            final_dict._set_internal_item(final_key, value)
+
+    @non_string_warning
+    def __getitem__(self, key):
+        """
+        Get single item
+
+        Parameters
+        ----------
+        key : str
+            key to desired item
+
+        Returns
+        -------
+        Any
+            value inside dict
+        """
+        if not isinstance(key, str) or '.' not in key:
+            try:
+                return super().__getitem__(int(key))
+            except (KeyError, ValueError):
+                return super().__getitem__(key)
+        else:
+            return self._traverse_keys(key.split("."), create=False)
+
+    @non_string_warning
+    def __contains__(self, key):
+        """
+        Check if key is in config
+        (also works for nested dicts with short form)
+
+        Parameters
+        ----------
+        key : str
+            key for desired value
+
+        Returns
+        -------
+        bool
+            true if key is in config
+        """
+        contain = True
+        try:
```

Comment:
Do we only want to resolve the key like this or with `nested_get` to also check it's presence in subdicts?

Reply:
We could add this to the `LookupConfig`. The default config does not support `nested_get`.